### ISSUE TITLE
Evaluate Profile-Guided Optimization (PGO)

### ISSUE THREAD

**Issue by zamazan4ik** [libsql/sqld#524]
Hi!

Did anyone already test `sqld` with Profile-Guided Optimization? I am curious since according to my tests, PGO helps to achieve better performance with SQLite: [link](https://sqlite.org/forum/forumpost/d26f4eba26) at least in different tests. Since `sqld` has many common with SQLite and other databases, I think these results can be useful for the project too.

More about PGO benefits to other apps (not only SQLite) you can find [here](https://github.com/zamazan4ik/awesome-pgo/).

**Comment by psarna** [comment]
Thanks for bringing it up! We haven't evaluated PGO yet, but it's a good idea, especially that database access usually has lots of common paths, even across different users.

Rust toolchain seems to have good support for it -- https://doc.rust-lang.org/rustc/profile-guided-optimization.html, and we can hook PGO into libSQL compilation as well.

/cc @penberg @MarinPostma

**Comment by zamazan4ik** [comment]
> Rust toolchain seems to have good support for it -- https://doc.rust-lang.org/rustc/profile-guided-optimization.html, and we can hook PGO into libSQL compilation as well.

Yep. For Rust projects, I recommend at first starting experimenting with PGO via https://github.com/Kobzol/cargo-pgo
